test/utils.cc

```diff
@@ -0,0 +1,33 @@
+#include "utils.hh"
+
+namespace blas {
+
+template <typename src_t, typename dst_t>
+void copy_matrix(
+    int m, int n,
+    src_t const* src, int ld_src,
+    dst_t*       dst, int ld_dst)
+{
+  #pragma omp parallel for collapse(2)
+  for (int i = 0; i < m; ++i) {
+    for (int j = 0; j < n; ++j) {
+      dst[i + j*ld_dst] = (dst_t)src[i + j*ld_src];
+    }
+  }
+}
```

Comment:
4-space indent.
`int64_t` as in CUDA above.
Usually I use C++ cast syntax: `dst_t( src[ ... ] )`. It's clearer which value is being cast. Consider the type of `(float) m * 2.0` vs `float( m * 2.0 )`.
For column-major access, the `i` loop should be the inner loop for better cache and vector efficiency.